gypbuild.js

```diff
@@ -28,6 +28,8 @@ function runGyp (opts, target, cb) {
     }
     if (opts.debug) args.push('--debug')
 
+    if (opts.format) args.push('--', '--format', opts.format)
```

Comment:
Why is `--` needed here? Isn't `--format` part of `node-gyp`'s arguments?

Reply:
I'm guessing `node-gyp` will take the arguments after `--` and pass them directly to the python code. I just don't see any docs on it.

Reply:
The logic in `parseArgv` is somewhat convoluted (https://github.com/nodejs/node-gyp/blob/master/lib/node-gyp.js#L135-L147), as far as I understood, the regular arguments (those before `--`) are consumed in that function while after `--` are passed to the python code.